cypress/e2e/audio-player/audio-player.spec.ts

```diff
@@ -0,0 +1,588 @@
+            percyCSS,
+            // 243px + 12px + 48px = 303px. See _EventBubbleTile.pcss and _AudioPlayer.pcss for spacing
+            widths: snapshotWidthsBubble,
+        });
+
+        cy.log("Took a snapshot of " + detail + " on bubble layout");
+
+        // Reset the layout
+        cy.setSettingValue("layout", null, SettingLevel.DEVICE, Layout.Group);
+    };
+
+    beforeEach(() => {
+        cy.startHomeserver("default").then((data) => {
+            homeserver = data;
+            cy.initTestUser(homeserver, TEST_USER).then(() =>
+                cy.createRoom({}).then((_roomId) => {
+                    roomId = _roomId;
+                }),
+            );
+        });
+
+        cy.injectAxe();
+    });
+
+    afterEach(() => {
+        cy.stopHomeserver(homeserver);
+    });
+
+    it("should be correctly rendered on IRC layout", () => {
+        // Please note that takeSnapshots() is not to be used here since it does not take snapshots on IRC layout.
+        // The design of it should be same as on modern layout.
+
+        visitRoom();
+
+        // Upload one second audio file with a long file name
+        uploadFile("cypress/fixtures/1sec-long-name-audio-file.ogg");
+
+        // Scroll to the bottom to make the audio player visible for Percy tests
+        cy.get(".mx_MainSplit .mx_ScrollPanel").scrollTo("bottom");
+
+        cy.get(".mx_RoomView_MessageList").within(() => {
+            checkPlayerFilenameLong();
+
+            // Enable IRC layout
+            cy.setSettingValue("layout", null, SettingLevel.DEVICE, Layout.IRC);
+
+            // Click the timestamp to highlight the event tile in case it is not visible
+            cy.get(".mx_EventTile_last[data-layout='irc'] .mx_MessageTimestamp").click();
+
+            // Assert that rendering of the player settled and the play button is visible before taking a snapshot
+            cy.get(".mx_MAudioBody [data-testid='play-pause-button'][aria-label='Play']").should("be.visible");
+
+            cy.get(".mx_MAudioBody").percySnapshotElement("Audio player (light theme) on IRC layout", {
+                percyCSS,
+                widths: snapshotWidthsIRC,
+            });
+
+            // Output a log for reference/debugging
+            cy.log("Took a snapshot of Audio player (light theme) on IRC layout");
+        });
+    });
+
+    it("should be correctly rendered on modern and bubble layouts", () => {
+        visitRoom();
+
+        // Upload one second audio file with a long file name
+        uploadFile("cypress/fixtures/1sec-long-name-audio-file.ogg");
+
+        // Scroll to the bottom to make the audio player visible for Percy tests
+        cy.get(".mx_MainSplit .mx_ScrollPanel").scrollTo("bottom");
+
+        cy.get(".mx_RoomView_MessageList").within(() => {
+            checkPlayerFilenameLong();
+
+            // Reset to the default layout
+            cy.setSettingValue("layout", null, SettingLevel.DEVICE, Layout.Group);
```

Comment:
is this necessary because we may still be in IRC mode from the previous test?

Reply:
No, this is not necessary. Thanks for spotting.

Reply:
This was removed with https://github.com/matrix-org/matrix-react-sdk/pull/10441/commits/1a0fde509b987d1d858e42f34e6956359b0bd2da.